packages/devtools_app/lib/src/extensions/embedded/_controller_web.dart

```diff
@@ -2,8 +2,92 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+// ignore_for_file: avoid_web_libraries_in_flutter, as designed
+import 'dart:async';
+import 'dart:html' as html;
+import 'dart:ui' as ui;
+
+import 'package:devtools_extensions/devtools_extensions.dart';
+
 import 'controller.dart';
 
+/// Incrementer for the extension iFrame view that will live for the entire
+/// DevTools lifecycle.
+///
+/// Each time [EmbeddedExtensionControllerImpl.init] is called, we create a new
+/// [html.IFrameElement] and register it to
+/// [EmbeddedExtensionControllerImpl.viewId] via
+/// [ui.platformViewRegistry.registerViewFactory]. Each new [html.IFrameElement]
+/// must have a unique id in the [PlatformViewRegistry], which
+/// [_viewIdIncrementer] is used to create.
+var _viewIdIncrementer = 0;
+
 class EmbeddedExtensionControllerImpl extends EmbeddedExtensionController {
-  // TODO(kenz): implement web controller for embedded extension
+  EmbeddedExtensionControllerImpl(super.extensionConfig);
+
+  /// The view id for the extension iFrame.
+  ///
+  /// See [_viewIdIncrementer] for an explanation of why we use an incrementer
+  /// in the id.
+  late final viewId =
+      'devtools-extension-${extensionConfig.name}-${_viewIdIncrementer++}';
```

Comment:
Are we going to send this view id in messages? If yes, how about decreasing size of the packages by shortening to something like

'ext-${extensionConfig.name}-${_viewIdIncrementer++}';

Reply:
we will not see this view id in messages, but shortened nonetheless.